lib/perl/Genome/Model/Build.pm

```diff
@@ -1395,6 +1395,11 @@ sub _launch {
     local $ENV{UR_DUMP_STATUS_MESSAGES} = 1;
     local $ENV{UR_COMMAND_DUMP_STATUS_MESSAGES} = 1;
 
+    # we don't use 'local' because we want this to persist through
+    # spawned workflows
+    $ENV{Genome::Config::spec('lsf_project_name')->env} =
+        sprintf("build/%s", $self->id);
```

Comment:
The `Scope::Guard` is only created if you ask for a return value from `set_env`, so I believe it should be used regardless.

My previous comment was referring to jobs that do belong to the parent build becoming incorrectly associated with the child thanks to this not being localized.  I agree nested builds are not great--unfortunately they're in `SomaticValidation` of all places...

I would guess the problem with localization is because the workflow is finally submitted in a commit observer.  The project variable needs to be set at that time to work properly.